compiler/parser/Parser.y

```diff
@@ -1150,6 +1153,31 @@ topdecl :: { LHsDecl GhcPs }
         -- it had a $(..) wrapped around it
         | infixexp_top                          { sLL $1 $> $ mkSpliceDecl $1 }
 
+-- Exceptions
+--
+exception_decl :: { OrdList (LHsDecl GhcPs) }
+  : 'exception' tycon arecord_with_opt exception_body_opt {% mkExceptionDecls $2 $3 $4 }
```

Comment:
I think I asked this before but I forgot: why do we limit exceptions to records?

Reply:
It's no less expressive than having variant exception types, enum excetpion types, etc, since you can always add a variant as a record field. From the perspective of the compiler it's kinda nice to have exceptions be uniform, but it also fits in better with the existing Daml syntax (e.g. for templates and choices). We also aren't really bound by this choice, we can always add variant exception types (or union exception types) in the future.